src/VisualStudio/Core/Def/ExternalAccess/VSTypeScript/Api/VSTypeScriptContainedLanguageWrapper.cs

```diff
@@ -40,6 +41,29 @@ internal struct VSTypeScriptContainedLanguageWrapper
                 vbHelperFormattingRule: null);
         }
 
+        public VSTypeScriptContainedLanguageWrapper(
+            IVsTextBufferCoordinator bufferCoordinator,
+            IComponentModel componentModel,
+            CodeAnalysis.Workspace workspace,
+            IVsHierarchy hierarchy,
+            uint itemid,
+            Guid languageServiceGuid)
+        {
+            var filePath = ContainedLanguage.GetFilePathFromHierarchyAndItemId(hierarchy, itemid);
+            var projectId = ProjectId.CreateNewId($"Project for {filePath}");
```

Comment:
do we need this new wrapper ctor here to access the file path?

Reply:
Discussed offline - it's to specify a workspace.